Add unary/binary ops necessary for mobilenet

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59051 Add peephole for len(x.size())
* #58932 Add more shape functions for mobilenet
* #58890 Switch symbolic shape registry to operator map'
* #56966 [JIT] Disable Complete Shape Inlining For Testing Purposes
* **#56828 Add unary/binary ops necessary for mobilenet**

Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

Commit:
Add unary/binary ops necessary for mobilenet

[ghstack-poisoned]

Commit:
Update on "Add unary/binary ops necessary for mobilenet"




[ghstack-poisoned]

Commit:
Update on "Add unary/binary ops necessary for mobilenet"




[ghstack-poisoned]

desertfire commented on torch/csrc/jit/runtime/symbolic_shape_registry.cpp:
```diff
@@ -41,52 +41,71 @@ const std::string shape_compute_functions =
+        def unary_two_unused_inputs(self: List[int], inp0: Any, inp1: Any):
+          out: List[int] = []
+          for elem in self:
+            out.append(elem)
```
Nit: You could use out.extend(self) instead of this loop.

eellison replied:
right now i am limiting it to operators which we can partially optimize out. in future prs im hoping to add more coverage to list ops and add hard errors when you use an operator that is not symbolically optimizable

desertfire commented on torch/csrc/jit/runtime/symbolic_shape_registry.cpp:
```diff
@@ -41,52 +41,71 @@ const std::string shape_compute_functions =
 
+        # TODO: maybe make it customary that extra arguments are unused ?
+        # TODO: return self directly
+        def unary_two_unused_inputs(self: List[int], inp0: Any, inp1: Any):
```
Just trying to understand the bigger picture here, does mean we need unary_one_unused_inputs, unary_three_unused_inputs, etc.? If those are unused inputs, why do we need to spell them out?

eellison replied:
Currently our symbolic shape analysis expects the number of inputs in the shape compute graph for an operator to have the same number of inputs as the operator. I think in a follow up i might make it so that it is implicit that extra arguments are unused.

desertfire reviewed (approved):
LGTM in general

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"




[ghstack-poisoned]

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"




[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).

eellison committed:
Update on "Add unary/binary ops necessary for mobilenet"


Differential Revision: [D28444660](https://our.internmc.facebook.com/intern/diff/D28444660)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28444660).
